packages/itmat-interface/src/graphql/resolvers/userResolvers.ts

```diff
@@ -34,6 +34,40 @@ export const userResolvers = {
             const queryObj = args.userId === undefined ? { deleted: null } : { deleted: null, id: args.userId };
             const cursor = db.collections!.users_collection.find(queryObj, { projection: { _id: 0 } });
             return cursor.toArray();
+        },
+        validateResetPassword: async (__unused__parent: Record<string, unknown>, args: any): Promise<IGenericResponse> => {
+            /* decrypt email */
+            if (args.token.length < 16) {
+                throw new ApolloError(errorCodes.CLIENT_MALFORMED_INPUT);
+            }
+            const salt = makeAESKeySalt(args.token);
+            const iv = makeAESIv(args.token);
+            let email;
+            try {
+                email = await decryptEmail(args.encryptedEmail, salt, iv);
+            } catch (e) {
+                throw new ApolloError('Token is not valid.');
+            }
+
+            /* check whether username and token is valid */
+            /* not changing password too in one step (using findOneAndUpdate) because bcrypt is costly */
+            const TIME_NOW = new Date().valueOf();
+            const ONE_HOUR_IN_MILLISEC = 60 /* minutes per hr */ * 60 /* sec per min */ * 1000 /* milli per unit */;
```

Comment:
This is not consistent with the same line below

Reply:
I simplified the validation process. As I think this part only needs to validate whether the link is expired, thus we do not need to validate other info, like password.

Reply:
I was more thinking about the inline-comments. The name of the variable is expressive enough that we don't need all the `/* ... */` 😄